### DISCUSSION TITLE
Define Process for Collaboration and Stages

### DISCUSSION THREAD

**Discussion by andorsk** [trustoverip/trust-spanning-protocol#6] [General]
Current proposal: 

How do we move forward as a group. What are the stages? What do they look like? 

IMO: This can move to a ticket and close once we have a PR that's merged for a CONTRIBUTING.md file, which (can) if required, reference a wiki.

```mermaid
graph TD
    Initiation -->|scoping, audience, deliverables defined| PS
    PS[Proposal Stage]
    CP[Consolidation Phase]
    DraftPhase[Draft Phase]
    PublicRelease[Public Release]
    PS -->|muliple proposals generated similar to aries-rfcs| CP
    CP -->|consolidate separate proposals into a unified proposal| DraftPhase
    DraftPhase -->|some repo deliverable for an actual proposal| PublicRelease
    PublicRelease --> CommunityFeedback[Community Feedback]
    CommunityFeedback  -->|feedback loop| PublicRelease
```

**Comment by brianorwhatever** [comment]
Should the feedback look after public release go through the same "RFC" process that the proposal stage uses? Are we planning on continuing to have an improvement process once this protocol exists in the world or will it be "set in stone" once it's mature

**Comment by talltree** [comment]
In the 2023-01-25 APAC call, we discussed the following points about the process:

- [Jo Spencer](https://github.com/JoOnGT) said that he felt we needed to go through a **modeling activity** first for the next layer down.
- Then we need to come to consensus on that model and test it out.
- This **modeling subgroup** can focus on this model and improve or change it as we learn more.

We also discussed Andor's version of the stages/milestones and agreed they made sense.

ACTION: @talltree (Drummond) will produce a second version with updates to the labels that correspond to the official names for certain stages according to the ToIP Foundation charter (based on our Joint Development Foundation legal roots at the Linux Foundation).

**Reply by talltree** [reply]
My action item is done — see [the post below](https://github.com/trustoverip/trust-spanning-protocol/discussions/6#discussioncomment-4876863).

**Comment by Oskar-van-Deventer** [comment]
As highlighted in https://github.com/trustoverip/trust-spanning-protocol/pull/7, the "initiation phase" should include what telco standardisation often calls "stage-1". That is, elaborations of use cases and requirements for the involved functional roles ("Alice", "Bob", "Service Provider", ...)

**Comment by talltree** [comment]
Per my action item above, here is a table defining the full set of work stages that a ToIP Working Group (WG) or Task Force (TF) would typically go through in developing a ToIP specification:

Stage | Activity | Exit Criteria
-- | -- | --
**Initiation** | Introduce members; agree on vision, mission, process, and tools | Consensus on vision, mission, process, tools
**Problem Definition** | Stakeholders propose use cases to build a map of what problems they need to solve for whom | Consensus on the problem map
**Requirements** | Extract and enumerate specific requirements from the problem map | Consensus on requirements
**Design Principles** | Develop the principles that need to guide/govern design of the solution | Consensus on design principles
**Proposals** | Members submit proposed solution designs | No further proposals
**Consolidation** | Members identify common elements and seek to develop a consolidated proposal | Consensus on contents of first Working Draft
**Working Drafts** † | A cycle of publishing Working Drafts, raising and resolving issues, and agreeing on revisions | Consensus on first Public Review Draft
**Public Review Drafts** | Same as Working Draft stage except with public review | Consensus (or vote) on WG Approved Draft
**WG Approved Draft** † | WG decision to submit for SC approval | Consensus (or vote) to submit for SC approval
**ToIP Approved Draft** † | SC decision to approve as ToIP Deliverable | Consensus (or vote) to approve

† _Stages officially recognized in the Linux Foundation Joint Development Foundation process._

Note: this table has been [published on the ToIP wiki](https://wiki.trustoverip.org/pages/viewpage.action?pageId=27724781).

**Reply by talltree** [reply]
Now, to clarify my comments in the first two TSPTF meetings, in our particular case, I believe we can skip the **Problem Definition**, **Requirements**, and **Design Principles** stages — and go straight to the **Proposals** stage — because 2.5 years of work has already been done on the overall problem definition (see [Introduction to ToIP](https://trustoverip.org/permalink/Introduction-to-ToIP-V2.0-2021-11-17.pdf)), requirements for the TSP (see the 18 requirements for Layer 2 in the [ToIP Technology Architecture V1.0 Specification](https://github.com/trustoverip/TechArch/blob/main/spec.md)), and design principles (see [Design Principles for the ToIP Stack](https://trustoverip.org/permalink/Design-Principles-for-the-ToIP-Stack-V1.0-2022-11-17.pdf)).

However, if any TSPTF members feel that we have not adequately addressed either of those three stages — for example, if you believe there are additional requirements or design principles that should be considered — you should feel free to put them forward using a Discussions thread. The [ToIP Technology Architecture V1.0 Specification](https://github.com/trustoverip/TechArch/blob/main/spec.md) is still in public review, so new or revised requirements for Layer 2 can be reflected there, and we are always open to publishing a second version of the [Design Principles for the ToIP Stack](https://trustoverip.org/permalink/Design-Principles-for-the-ToIP-Stack-V1.0-2022-11-17.pdf).

**Reply by Oskar-van-Deventer** [reply]
Hi Drummond (@talltree), I see that I need some reading up. It is always challenging to enter an ongoing standards development process as a "latecomer", and I'd definitely would not want to hold up the process. I shall use the Discussions thread if/when appropriate. Thanks for bringing together references to all relevant materials in one single post!

Oskar
